File: algorithms/2/Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--",
                 "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
        alpha = 'abcdefghijklmnopqrstuvwxyz'
        dict1 = dict(zip(alpha, morse))
        print(dict1)
        s = ''
        set1 = set()
        for ele in words:
            s = ''.join(map(lambda ele: dict1[ele], ele))
            set1.add(s)

        return len(set1)

File: algorithms/2/Unique Morse Code Words.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    int uniqueMorseRepresentations(vector<string> &words)
    {
        string morse[] = {".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."};
        string alpha = "abcdefghijklmnopqrstuvwxyz";

        unordered_map<char, string> map;

        for (int i = 0; i < 26; i++)
        {
            map[alpha[i]] = morse[i];
        }
        set<string> set1;
        for (auto ele : words)
        {
            string result = "";
            for (auto ele1 : ele)
            {
                result+=map[ele1];
            }
            set1.insert(result);
        }
        return set1.size();
    }
};